specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/workspaceRP.json

```diff
@@ -3138,12 +3202,27 @@
           "description": "Category of the connection",
           "$ref": "#/definitions/ConnectionCategory"
         },
+        "createdByWorkspaceArmId": {
+          "type": "string"
+        },
         "expiryTime": {
           "format": "date-time",
           "type": "string"
         },
+        "isSharedToAll": {
```

Comment:
>isSharedToAll [](http://example.com/codeflow?start=9&length=13)

booleans are not future proof. And a shared user list doesn't sound very future proof either. This seems to be an Authorization scenario. Why isn't it being handled with RBAC?

Reply:
Long story short, this can't achieved by RBAC due to what we want to achieve, we will have to do RBAC office hour before we implement the shared user list logic, but since we are currently working on it we wish to have this in swagger to unblock the upcoming SDK development.

Reply:
Can you please provide more details as why this is not possible using RBAC today? I understand the quick deliverables, but we shouldn't leave loopholes with respect to Auth related workflows.

Reply:
Sure, so basically we have a "kind" on the workspace resource level, and there is a parent-child related between the "Hub" kind and "Project" kind, there can be many "Project" created based on a "Hub", and the property here is to make sure some connection resource created on one "Project" can be shared to other Project created based on the same "Hub", the userList is to control whether we want to share the connection to the Project created by that user under the same "Hub".

Since RBAC will not have knowledge of the relation between Project and Hub, we have to do this share logic on the service level, so this is not like we change the accessibility of the resource itself, it is used to tell the service whether we should copy some of the configurations to the resource user want to share.

so this is something like RBAC but can't achieve through RBAC as we want to limit the share within the "Hub" scope which is not achievable through RBAC.

Reply:
Service level custom RBAC seems dangerous. Why are not the hubs and projects modeled with ARM Resources with actual parent child relationship. Please attend ARM API Modeling office hours and discuss your scenario to get this reviewed.

Reply:
OK thanks, I will remove the sharedUserList and attend the office hours to get feedback before implementing that, for the parent-child relation we already did ARM office hour review on that and we didn't do it on the ARM level is that will introduce too much breaking change no outside, and we got approved when choosing this way of implementation. So there is not much we can do on that part.

Reply:
Got more feedback about this, so basically this is not a RBAC, this is a sharing feature, like users can choose a folder to share in the domain, RBAC is related on who can use the connection. That's controlled by Azure RBAC